VDA-XP-Reward/test/index.test.ts

```diff
@@ -0,0 +1,844 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { SnapshotRestorer, takeSnapshot, time } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { Wallet } from "ethers";
+import { anyValue } from "@nomicfoundation/hardhat-chai-matchers/withArgs";
+
+import hre, { ethers , upgrades } from "hardhat"
+import { VDAXPReward } from "../typechain-types";
+import { VeridaToken } from "@verida/erc20-contract/typechain-types";
+import { VeridaDIDRegistry } from "@verida/did-registry-contract/typechain-types";
+import EncryptionUtils from '@verida/encryption-utils'
+
+import { abi as TokenABI, bytecode as TokenByteCode } from "@verida/erc20-contract/artifacts/contracts/VDA-V1.sol/VeridaToken.json";
+import { abi as DIDRegistryABI, bytecode as DIDRegistryByteCode } from "@verida/did-registry-contract/artifacts/contracts/VeridaDIDRegistry.sol/VeridaDIDRegistry.json";
+
+import { endPoints, getRegisterSignature } from "./didHelper";
+import { CLAIM_INVALID_XP, CLAIM_GAMER31, ClaimData, ClaimInfo, CLAIM_ZKPASS } from "./claimHelper";
+
+let accountList: SignerWithAddress[];
+let owner: SignerWithAddress;
+let user: SignerWithAddress;
+const registeredDID = Wallet.createRandom();
+
+const trustedSigners = [
+    Wallet.createRandom(),
+    Wallet.createRandom(),
+    Wallet.createRandom(),
+]
+
+// Reward receives address
+const receiverAddress = [
+    Wallet.createRandom().address,
+    Wallet.createRandom().address,
+    Wallet.createRandom().address,
+]
+
+describe("VeridaXPReward", () => {
+    let contract: VDAXPReward;
+    let token: VeridaToken;
+    let didRegistry: VeridaDIDRegistry;
+    let veridaWallet: SignerWithAddress;
+
+    const deployContracts = async() => {
+        // Deploy and initialize token
+        const tokenFactory = await ethers.getContractFactory(TokenABI, TokenByteCode)
+        token = await tokenFactory.deploy() as VeridaToken
+        await token.deployed()
+        await token.initialize();
+
+        await token.enableTransfer();
+
+        // Deploy and initialize DIDRegistry
+        const didRegistryFactory = await ethers.getContractFactory(DIDRegistryABI, DIDRegistryByteCode);
+        didRegistry = await didRegistryFactory.deploy() as VeridaDIDRegistry
+        await didRegistry.deployed();
+        
+
+        const contractFactory = await ethers.getContractFactory("VDAXPReward")
+        contract = (await upgrades.deployProxy(
+            contractFactory,
+            [token.address, didRegistry.address],
+            {
+                initializer: '__VDAXPReward_init'
+            }
+        )) as VDAXPReward
+        await contract.deployed()
+    }
+
+    before(async() => {
+        accountList = await ethers.getSigners();
+        owner = accountList[0]
+        user = accountList[1]
+        veridaWallet = accountList[2];
+        
+        await deployContracts()
+    })
+
+    describe("TrustedSigners", () => {
```

Comment:
We can remove these tests as they already existing in the VDAVerificationBase library

Reply:
Though it was tested in the `VDAVerificationBase`, it might better to double check this `XPReward` contract support that.
Would you like to remove this code?

Reply:
Ok, let's leave then.